src/watchdog/observers/read_directory_changes.py

```diff
@@ -58,8 +58,10 @@ def start(self) -> None:
             sleep(0.01)
 
     def on_thread_stop(self) -> None:
-        if self._whandle:
-            close_directory_handle(self._whandle)
+        whandle = self._whandle
+        if whandle:
+            self._whandle = None
```

Comment:
I'm not sure I understand what race this is guarding against, but it seems like it would be better to have a lock around this.

Reply:
This is guarding against the file handle re-use crash described [here](https://github.com/gorakhargosh/watchdog/issues/1132#issuecomment-3307704051). Let me see if locking also works, since that's a lot more explicit.